Resources/Prototypes/Entities/Objects/Weapons/Guns/Battery/battery_guns.yml

```
@@ -713,3 +713,47 @@
   - type: Appearance
   - type: StaticPrice
     price: 750
+
+- type: entity
+  name: energy shotgun
+  parent: [BaseWeaponBattery, BaseGunWieldable]
+  id: WeaponEnergyShotgun
+  description: An one of a kind prototype energy weapon using various shotgun configurations. The possibility for both lethal, and non-lethal shots makes this a versatile weapon. 
```

Some flavor text changes I'd like to offer:

I think that looks better

```suggestion
  description: A one-of-a-kind prototype energy weapon that uses various shotgun configurations. It offers the possibility of both lethal and non-lethal shots, making it a versatile weapon.
```
This version is closer to yours, just fixes some typos here and there
```suggestion
  description: A one-of-a-kind prototype energy weapon that uses various shotgun configurations. The possibility of both lethal and non-lethal shots makes it a versatile weapon.
```